non-critical-infra/modules/postfix.nix

```diff
@@ -0,0 +1,48 @@
+{ config, pkgs, ... }:
+
+{
+  sops.secrets.opendkim-private-key = {
+    sopsFile = ../secrets/opendkim-private-key.caliban;
+    format = "binary";
+    owner = config.services.postfix.user;
+  };
+  services.opendkim = {
+    enable = true;
+    domains = config.networking.fqdn;
+    selector = "mail";
+    user = config.services.postfix.user;
+    group = config.services.postfix.group;
+    keyPath = "/run/opendkim-keys";
+  };
+
+  systemd.services.opendkim.serviceConfig = {
+    ExecStartPre = [
+      (
+        "+${pkgs.writeShellScript "opendkim-keys" ''
+          install -o ${config.services.postfix.user} -g ${config.services.postfix.group} -D -m0700 ${config.sops.secrets.opendkim-private-key.path} /run/opendkim-keys/${config.services.opendkim.selector}.private
+        ''}"
+      )
+    ];
+  };
+
+  services.postfix = {
+    enable = true;
+    hostname = config.networking.fqdn;
+    domain = config.networking.fqdn;
+    config = {
+      smtp_tls_note_starttls_offer = "yes";
+      smtp_tls_security_level = "dane";
```

Comment:
```
postfix/smtp[213546]: warning: DNSSEC validation may be unavailable
postfix/smtp[213546]: warning: reason: dnssec_probe 'ns:.' received a response that is not DNSSEC validated
```

We'd probably need to enable DNSSEC validation in resolved or set up a dedicated recursor. Not worth the hassle IMO.

Reply:
I think the proper downgrade here would be `"may"`.

https://www.postfix.org/postconf.5.html#smtp_tls_security_level